samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -978,6 +977,9 @@ class SamzaContainer(
         case unknownException: Throwable =>
           warn("Received an exception when persisting locality info for container %s: " +
             "%s" format (containerContext.id, unknownException.getMessage))
+      } finally {
+        info("Shutting down locality manager.")
+        localityManager.close()
```

Comment:
> Possible to move LocalityManager construction in this method as well?

Sounds like a good idea, done.

> Also, do you think it'll be useful if containers always reported their locality, regardless of whether host affinity is on or off?

Currently in YARN model, when host affinity is turned off,  non-sticky `ContainerAllocator` is used(Even if we report the host assignment, the `ContainerAllocator` won't use it for container-to-host assignment). In case of standalone, the processors register themselves to the group with their locality information. I think it might be unnecessary to report host assignment of containers by default.